pkg/resource/labels.go

```diff
@@ -27,28 +32,130 @@ func (labels Labels) Equal(other Labels) bool {
 
 // Matches if labels match the LabelTerm.
 func (labels Labels) Matches(term LabelTerm) bool {
-	if labels.KV.Empty() {
-		return term.Op == LabelOpNotExists
+	matches := labels.matches(term)
+
+	if term.Invert {
+		return !matches
 	}
 
-	switch term.Op {
-	case LabelOpNotExists:
-		_, ok := labels.Get(term.Key)
+	return matches
+}
 
-		return !ok
-	case LabelOpExists:
-		_, ok := labels.Get(term.Key)
+func (labels Labels) matches(term LabelTerm) bool {
+	if labels.KV.Empty() && term.Op == LabelOpExists {
+		return false
+	}
+
+	value, ok := labels.Get(term.Key)
 
-		return ok
+	if !ok {
+		return false
+	}
+
+	switch term.Op {
+	case LabelOpExists:
+		return true
 	case LabelOpEqual:
-		value, ok := labels.Get(term.Key)
+		return value == term.Value
+	case LabelOpIn:
+		return slices.Contains(term.Set, value)
+	case LabelOpLTE:
+		return value <= term.Value
+	case LabelOpLT:
+		return value < term.Value
+	case LabelOpLTNumeric:
+		left, right, ok := getNumbers(value, term.Value)
+		if !ok {
+			return false
+		}
 
+		return left < right
+	case LabelOpLTENumeric:
+		left, right, ok := getNumbers(value, term.Value)
 		if !ok {
 			return false
 		}
 
-		return value == term.Value
+		return left <= right
 	default:
 		panic(fmt.Sprintf("unsupported label term operator: %v", term.Op))
 	}
 }
+
+var (
+	re    = regexp.MustCompile(`^(\d+)(.*)$`)
+	units = map[string]float64{
+		"pi": math.Pow(1024, 5),
+		"ti": math.Pow(1024, 4),
+		"gi": math.Pow(1024, 3),
+		"mi": math.Pow(1024, 2),
+		"ki": 1024,
+
+		"p": math.Pow(10, 15),
+		"t": math.Pow(10, 12),
+		"g": math.Pow(10, 9),
+		"m": math.Pow(10, 6),
+		"k": math.Pow(10, 3),
+	}
+)
+
+func getNumbers(left, right string) (int64, int64, bool) {
+	numLeft, ok := parseValue(left)
+	if !ok {
+		return 0, 0, false
+	}
+
+	numRight, ok := parseValue(right)
+	if !ok {
+		return 0, 0, false
+	}
+
+	return numLeft, numRight, true
+}
+
+func parseValue(value string) (int64, bool) {
```

Comment:
it will be cool to extract this function and related to a small `internal/` sub-package, and have it covered with proper tests